storm-server/src/main/java/org/apache/storm/scheduler/resource/ResourceUtils.java

```diff
@@ -189,13 +223,18 @@ private static void checkInitCpu(Map<String, Double> topologyResources, String c
                         null);
                     topologyResources.put(Config.TOPOLOGY_COMPONENT_CPU_PCORE_PERCENT, topoCpu);
                 }
-                LOG.debug("Topology Resources {}", topologyResources);
+
+                // If resource is also present in resources map will overwrite the above
+                if (jsonObject.containsKey(Config.TOPOLOGY_COMPONENT_RESOURCES_MAP)) {
+                    topologyResources.putAll((Map<String, Double>) jsonObject.get(Config.TOPOLOGY_COMPONENT_RESOURCES_MAP));
+                }
+                    LOG.info("Topology Resources {}", topologyResources);
```

Comment:
Is this needed?  Should it be debug if it is needed? and can we fix the indentation?